### PR TITLE
❇️  Add selectors for dark mode and locales, also update all the styles for white mode

### PR DESCRIPTION
**Pull request by Ked57**
image
image
image
image
image
image

### CODE REVIEW

File: src/utils/darkModeContext.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { createContext } from "react";
+
+type DarkModeContext = {
+  darkMode: boolean;
+  changeDarkMode: (darkMode: boolean) => void;
+};
+
+export const DarkModeContext = createContext<DarkModeContext>(undefined);
```
Comment (hgwood) [inline src/utils/darkModeContext.ts:8]: The type of the context value is `DarkModeContext` which does not accept `undefined` but the initial value is `undefined`? Sounds fishy. :)

File: src/pages/_app.tsx [hunk 2]
```diff
@@ -16,11 +17,45 @@ if (!BASE_URL) {
...
 const App = ({ Component, pageProps }) => {
   const { push, pathname: pathName, asPath, locale } = useRouter();
+  const storedDarkModeString = process.browser
+    ? window.localStorage.getItem("darkMode")
+    : true;
+  const storedLocaleString = process.browser
+    ? window.localStorage.getItem("locale")
+    : locale;
+
+  const storedDarkMode = Boolean(storedDarkModeString === "true");
```
Comment (hgwood) [inline src/pages/_app.tsx:27]: Doesn't `===` already return a boolean?

**Review by hgwood — COMMENTED** [review]
I think this could be cleaner, using [this technique](https://kentcdodds.com/blog/how-to-use-react-context-effectively), which extracts the building of the context value to a custom provider, and the usage of the context to a custom hook.

File: src/pages/_app.tsx (hunk 2 above)
Comment (Ked57) [reply src/pages/_app.tsx:27]: Yes I forgot to remove the Boolean stuff 😄

File: src/utils/darkModeContext.ts (hunk 1 above)
Comment (Ked57) [reply src/utils/darkModeContext.ts:8]: You're right i'll update this

**Review by hgwood — APPROVED** [review]
👍

**Pull request merged**